feat: added transcription language parameter

e-fominov opened the pull request:
By default, whisper-assistant-vscode translates everything into English which is perfectly fine for most use cases, but some users may want to use their native language, or translate into a different language.
This PR adds `whisper-assistant.transcriptionLanguage` setting with default to `en` (default behavior unchanged). User now can make it empty leading to using the spoken language transcription without a translation, use use any language supported by Whisper.

e-fominov committed:
added transcription language parameter

martin-opensky commented on package.json:
```diff
@@ -62,6 +62,11 @@
+        "whisper-assistant.transcriptionLanguage": {
+          "type": "string",
+          "default": "en",
+          "description": "Translate all the recordings into selected language. If empty, then auto-detect language and use it."
```
Could update so that the option is a list of whisper supported languages, rather than a free type input?  Also the default should be 'en' instead of auto-detect.

e-fominov replied:
updated package.json to have language selection enum

martin-opensky commented on src/speech-transcription.ts:
```diff
@@ -299,7 +299,7 @@ class SpeechTranscription {
         file: audioFile,
         model: model,
-        language: 'en',
+        language: config.get<string>('transcriptionLanguage'),
```
add ?? 'en' as a backup

e-fominov replied:
added ?? 'English'

martin-opensky commented:
Thanks for your PR @e-fominov.  I listed a couple of minor changes.  Could you also pull main into your branch as another PR has been merged which is causing a conflict?

e-fominov committed:
Transcription language selection enum setting

e-fominov committed:
use English as default

e-fominov commented:
updated PR to include language selection enum. using languages from https://github.com/openai/whisper/blob/main/whisper/tokenizer.py#L10-L110 with keeping the origin Whisper language order (popularity/quality/amount of training data).
Enum is made with full language names, while Whisper requires input language parameter to be [ISO 639-1](https://wikipedia.org/wiki/ISO_639-1) codes. Added LANGUAGE_CODES map to translate between language names into ISO codes.
If "Auto" is selected, then there is no value in LANGUAGE_CODES map leading to undefined language code, which is correct for Whisper making it automatically detect the language.

Please keep in mind, this is not a real translation between languages but a language detection hint. If one selects "Korean" language but speaks English, he is expected to get English transcription.

Pull request has been rebased  to be on the tip of master branch.
